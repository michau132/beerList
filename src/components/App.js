import React from 'react';
import { Provider } from 'mobx-react';
import { ThemeProvider } from 'styled-components';
import {
  HashRouter,
  Route,
} from 'react-router-dom';
import { configure } from 'mobx';
import Layout from '../layout/Layout';
import theme from '../utils/theme';
import stores from '../stores';
import ModalSwitch from '../containers/ModalSwitch';

configure({ enforceActions: 'always' });

const App = () => (
  <Provider {...stores}>
    <HashRouter>
      <ThemeProvider theme={theme}>
        <Layout>
          <Route component={ModalSwitch} />
        </Layout>
      </ThemeProvider>
    </HashRouter>
  </Provider>
);


export default App;
