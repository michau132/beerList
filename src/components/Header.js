import React from 'react';
import styled from 'styled-components';

const H1 = styled.h1`
  font-size: 40px;
  color: #fdc22a;
  text-transform: uppercase;
  margin-left: 40px;
  margin-bottom: 30px;
  span {
    color: #b6b6b6;
  }
`;

const Header = () => (
  <H1>
      beer
    <span>list</span>
  </H1>
);

export default Header;
