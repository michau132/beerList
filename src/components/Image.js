import React from 'react';
import PropTypes from 'prop-types';
import styled, { keyframes } from 'styled-components';
import LazyLoad, { forceCheck } from 'react-lazyload';

const spin = keyframes`
  0% { transform: rotate(0deg); }
  100% { transform: rotate(360deg); }
`;
const LoadingSpinner = styled.div`
  position: relative;
  height: 100%;
  width: 100%;
  display: flex;
  justify-content: center;
  align-items: center;
  
  &::after {
    content: '';
    border: 16px solid #f3f3f3;
    border-radius: 50%;
    border-top: 16px solid ${({ theme }) => theme.color.title};
    width: 50px;
    height: 50px;
    animation: ${spin} 2s linear infinite;
  }
`;

const ImageStyles = styled.img`
  object-fit: contain;
  max-height: 100%;
  max-width: 100%;
  height: 100%;
  width: 100%;
`;

const Image = ({
  src, alt, overflow, modal,
}) => {
  if (modal) {
    forceCheck();
  }
  return (
    <LazyLoad
      overflow={overflow}
      height={180}
      offset={100}
      once
      placeholder={<LoadingSpinner />}
      debounce={500}
    >
      <ImageStyles src={src} alt={alt} data-test="image" />
    </LazyLoad>
  );
};

Image.defaultProps = {
  overflow: false,
  modal: false,
};

Image.propTypes = {
  src: PropTypes.string.isRequired,
  alt: PropTypes.string.isRequired,
  overflow: PropTypes.bool,
  modal: PropTypes.bool,
};

export default Image;
