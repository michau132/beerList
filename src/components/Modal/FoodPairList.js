import React from 'react';
import styled from 'styled-components';
import { inject, PropTypes } from 'mobx-react';
import FoodPairListItem from './FoodPairListItem';

const FoodPairListStyle = styled.ul`
  list-style: none;
  padding: 0;
  margin-bottom: 10px;
  
  ${({ theme }) => theme.media.desktop} {
    margin-bottom: 20px;
  }
`;

const FoodPairList = ({
  ModalStore: {
    beer: {
      food_pairing,
    },
  },
}) => (
  <FoodPairListStyle>
    <h4>Best served with:</h4>
    {
      food_pairing.map(item => (
        <FoodPairListItem key={item} text={item} />
      ))
    }
  </FoodPairListStyle>
);

FoodPairList.propTypes = {
  ModalStore: PropTypes.observableObject.isRequired,
};

export default inject('ModalStore')(FoodPairList);
