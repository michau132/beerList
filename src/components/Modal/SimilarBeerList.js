import React from 'react';
import styled from 'styled-components';
import { inject, PropTypes } from 'mobx-react';
import SimilarBeerListItem from './SimilarBeerListItem';

const BeerListTitle = styled.h3`
  margin-bottom: 10px;
`;

const BeerListCnt = styled.ul`
  display: flex;
  flex-direction: column;
  padding: 0;
  list-style: none;
  
  ${({ theme }) => theme.media.tablet} {
    flex-direction: row;
  }
`;


const SimilarBeerList = ({ ModalStore: { similarBeers } }) => (
  <div>
    <BeerListTitle>You might also like</BeerListTitle>
    <BeerListCnt>
      {
        similarBeers.map(beer => (
          <SimilarBeerListItem key={beer.id} {...beer} />
        ))
      }
    </BeerListCnt>
  </div>
);

SimilarBeerList.propTypes = {
  ModalStore: PropTypes.observableObject.isRequired,
};

export default inject('ModalStore')(SimilarBeerList);
