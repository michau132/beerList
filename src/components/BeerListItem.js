/* eslint-disable react/no-unused-prop-types */
/* eslint-disable react/prop-types */
import React from 'react';
import styled from 'styled-components';
import { withRouter } from 'react-router';
import { compose } from 'recompose';
import { observer, inject } from 'mobx-react';
import Image from './Image';

const BeerListItemStyle = styled.li`
  padding: 20px;
  background-color: #fff;
  box-shadow: 0px 0px 4px 1px rgba(122,122,122,0.9);
  transition: .5s all;
  text-align: center;
  margin: 20px;
  width: 320px;
  cursor: pointer;
  
  :hover {
    transform: scale(1.05);
    box-shadow: 0px 0px 29px 9px rgba(122,122,122,1);
  }
`;

const ImgWrapper = styled.div`
  height: 180px;
  margin-bottom: 10px;
`;

const BeerName = styled.div`
  font-size: 24px;
  color: ${({ theme }) => theme.color.title};
  font-weight: bold;
`;

const BeerDesc = styled.div`
  font-size: 18px;
  color: #b7b7b7;
`;

const BeerListItem = ({
  beer, ModalStore, history,
}) => {
  const {
    id, image_url, name, tagline,
  } = beer;
  const handleClick = () => {
    history.push({
      pathname: `/${id}`,
      state: { modal: true },
    });
    ModalStore.showBeerDetails(beer);
  };
  return (
    <BeerListItemStyle
      onClick={handleClick}
    >
      <div>
        <ImgWrapper>
          <Image src={image_url} alt={name} overflow />
        </ImgWrapper>
      </div>
      <BeerName>{name}</BeerName>
      <BeerDesc>
        {tagline}
      </BeerDesc>
    </BeerListItemStyle>
  );
};

// BeerListItem.propTypes = {
//   id: PropTypes.number.isRequired,
//   image_url: PropTypes.string.isRequired,
//   name: PropTypes.string.isRequired,
//   tagline: PropTypes.string.isRequired,
// };

export default compose(
  inject('ModalStore'),
  observer,
  withRouter,
)(BeerListItem);
