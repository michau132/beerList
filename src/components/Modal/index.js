/* eslint-disable react/prop-types */
import React from 'react';
import { compose } from 'recompose';
import { Spring } from 'react-spring';
import BeerDetails from './BeerDetails';
import SimilarBeerList from './SimilarBeerList';
import FoodPairList from './FoodPairList';
import withLoading from '../../hoc/withLoading';
import withError from '../../hoc/withError';


export const Modal = () => (
  <Spring
    from={{ height: 400, transform: 'scale3d(0.5, 0.5, 0.5)' }}
    to={{ height: 'auto', transform: 'scale3d(1, 1, 1)' }}
    leave={{ height: 400, transform: 'scale3d(0.1, 0.1, 0.1)' }}
    config={key => (key === 'transform' ? { delay: 200 } : {})}
  >
    { (props => (
      <div style={props}>
        <BeerDetails />
        <FoodPairList />
        <SimilarBeerList />
      </div>
    ))}

  </Spring>
);


// Modal.propTypes = {
//   beer: PropTypes.shape({
//     image_url: PropTypes.string.isRequired,
//     name: PropTypes.string.isRequired,
//     tagline: PropTypes.string.isRequired,
//     ibu: PropTypes.number,
//     abv: PropTypes.number,
//     ebc: PropTypes.number,
//     description: PropTypes.string.isRequired,
//     food_pairing: PropTypes.arrayOf(PropTypes.string).isRequired,
//   }).isRequired,
//   similarBeers: PropTypes.arrayOf(PropTypes.shape({
//     id: PropTypes.number.isRequired,
//     image_url: PropTypes.string.isRequired,
//     name: PropTypes.string.isRequired,
//   })).isRequired,
// };


export default compose(withLoading, withError)(Modal);
