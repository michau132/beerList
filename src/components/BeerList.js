import React, { Fragment } from 'react';
import styled from 'styled-components';
import { compose } from 'recompose';
import { inject, observer, PropTypes } from 'mobx-react';
import withLoading from '../hoc/withLoading';
import withError from '../hoc/withError';
import BeerListItem from './BeerListItem';
import LastBeer from '../hoc/lastBeer';

const BeerListStyle = styled.ul`
  display: flex;
  justify-content: center;
  flex-wrap: wrap;
  padding: 0;
  list-style: none;
  padding-bottom: 150px;
  margin-bottom: 30px;
`;

const BeerList = ({ BeerListStore: { beerList } }) => (
  <BeerListStyle>
    {
        beerList.map(beer => (
          <Fragment key={beer.id}>
            <BeerListItem
              beer={beer}
            />
          </Fragment>
        ))
    }
  </BeerListStyle>
);

BeerList.propTypes = {
  BeerListStore: PropTypes.observableObject.isRequired,
};

export default compose(
  inject('BeerListStore'),
  observer,
  withLoading,
  withError,
  LastBeer,
)(BeerList);
