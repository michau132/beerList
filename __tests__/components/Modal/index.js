import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import { Modal } from '../../../src/components/Modal';

describe('<Modal /> component', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<Modal />).dive();
  });

  test('renders without crashing', () => {
    expect(wrapper).toBeDefined();
  });

  test('matching snapshot', () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
