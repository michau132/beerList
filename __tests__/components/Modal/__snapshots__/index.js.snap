// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Modal /> component matching snapshot 1`] = `
<div
  style={
    Object {
      "height": "auto",
      "position": "absolute",
      "transform": "scale3d(1, 1, 1)",
      "visibility": "hidden",
    }
  }
>
  <inject-BeerDetails-with-ModalStore />
  <inject-FoodPairList-with-ModalStore />
  <inject-SimilarBeerList-with-ModalStore />
</div>
`;
